src/zjs_modules.c

```diff
@@ -139,6 +141,39 @@ static jerry_value_t native_require_handler(const jerry_value_t function_obj,
     }
     DBG_PRINT("Native module not found, searching for JavaScript module %s\n",
               module);
+#ifdef ZJS_LINUX_BUILD
+    // Linux can pass in the script at runtime, so we have to read in/parse any
+    // JS modules now rather than at compile time
+    char full_path[size + 8];
+    char* str;
+    uint32_t len;
+    memcpy(full_path, "modules/", 8);
+    memcpy(full_path + 8, module, size);
+    full_path[size + 8] = '\0';
```

Comment:
This writes past the end of the array. `sprintf` might be a good option here, too. `zjs_copy_string` guarantees a null-terminated `module`.